src/setup/run.go

```diff
@@ -145,6 +146,11 @@ func ProvisionFunctionsServerlessAWS(config *Configuration, serverlessDirPath st
 }
 
 func ProvisionFunctionsServerlessAzure(config *Configuration, serverlessDirPath string) {
+	mu := sync.Mutex{}
+	wg := sync.WaitGroup{}
+
+	randomExperimentTag := util.GenerateRandLowercaseLetters(5)
```

Comment:
What is the purpose of the randomized tag? Is parallelism and subexperiment number not sufficient to differentiate?

Reply:
Previously I faced an issue where two experiments running concurrently (e.g. two end to end tests for a PR) were both trying to interact with the same resource group (like `sls-wus-dev-stellar-azure-subex0-para0-rg`)

Reply:
Makes sense!